ksql-streams/src/main/java/io/confluent/ksql/execution/streams/StreamStreamJoinBuilder.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.streams;
+
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.execution.builder.KsqlQueryBuilder;
+import io.confluent.ksql.execution.context.QueryContext;
+import io.confluent.ksql.execution.plan.Formats;
+import io.confluent.ksql.execution.plan.StreamStreamJoin;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.ksql.PhysicalSchema;
+import io.confluent.ksql.serde.KeySerde;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.streams.kstream.JoinWindows;
+import org.apache.kafka.streams.kstream.Joined;
+import org.apache.kafka.streams.kstream.KStream;
+
+public final class StreamStreamJoinBuilder {
+  private static final String LEFT_SERDE_CTX = "left";
+  private static final String RIGHT_SERDE_CTX = "right";
+
+  private StreamStreamJoinBuilder() {
+  }
+
+  public static KStream<Object, GenericRow> build(
```

Comment:
Generics are wrong again.  This should be `KStream<?, GenericRow>`, which again is unfortunately giving up any type information on the key :(

Really, the signature of this should be:

```java
public static <K>  KStream<K, GenericRow> build(
     final KStream<K, GenericRow> left,
      final KStream<K, GenericRow> right,
      final StreamStreamJoin join,
      final KsqlQueryBuilder queryBuilder,
      final JoinedFactory joinedFactory
) {
```